composer/trainer/trainer.py

```diff
@@ -888,6 +894,15 @@ def __init__(
             precision = Precision(precision)
         _validate_precision(precision, device)
 
+        # Determine whether torch compile is enabled
+        is_torch_2_0 = using_torch_2_0()
+        if is_torch_2_0 and use_torch_compile:
+            model = torch.compile(model)  # pyright: ignore
+        elif not is_torch_2_0 and use_torch_compile:
+            raise ValueError(f'torch.compile() is supported for PyTorch version >= 2.x.' +
```

Comment:
In the commit https://github.com/mosaicml/composer/pull/2118/commits/c4517821736fa5446ff1d2c415e97c70156b6ca2, I have converted to a warning message since we do not want to break the flow if user by mistakenly provided `compile_config` for torch < 2.0